cmd/mobynit/main.go

```diff
@@ -56,6 +49,22 @@ func mountContainer(containerID, graphDriver string) string {
 	if err := unix.Mount("", newRootPath, "", unix.MS_REMOUNT, ""); err != nil {
 		log.Fatal("error remounting container as read/write:", err)
 	}
+
+	return newRootPath
+}
+
+func prepareForPivot(sysroot, containerID, graphDriver string) string {
+	if err := os.MkdirAll("/dev/shm", os.ModePerm); err != nil {
+		log.Fatal("creating /dev/shm failed:", err)
+	}
+
+	if err := unix.Mount("shm", "/dev/shm", "tmpfs", 0, ""); err != nil {
+		log.Fatal("error mounting /dev/shm:", err)
+	}
+	defer unix.Unmount("/dev/shm", unix.MNT_DETACH)
+
+	newRootPath := mountContainer(filepath.Join(sysroot, LAYER_ROOT), containerID, graphDriver)
```

Comment:
I doesn't look right that both main() and this function call `mountContainer` with the exact same parameters. Can't this be done in a single place and just differentiate before and after this call based on whether we're doing a normal mount or we're booting?

Reply:
Mounting the container before init requires remounting read-write, shm tmpfs mounts then mountCountainer, etc. While a container using sysroot doesn't require it.

While adding multiple if/else checks, I restructured quite a bit. But still doesn't feel elegant. I don't see a way to make this elegant and muxed together. Having 2 obviously separate code paths like this current commit here look more readable.

You can see the other way here if you want https://github.com/ZubairLK/balena/blob/a83b228e94062fb719a7c52f975b80b0ab5f857c/cmd/mobynit/main.go